tl.tl

```diff
@@ -4663,6 +4715,26 @@ parse_statements = function(ps: ParseState, i: integer, toplevel?: boolean): int
       i, item = fn(ps, i)
 
       if item then
+         -- store unattached comments
+         if toplevel and token.comments then
+            for _, tc in ipairs(token.comments) do
```

Comment:
Perhaps make this into a separate function, like so?

```lua
if toplevel and token.comments then
   unattached_comments = store_unattached_comments(unattached_comments, token, item)
end
```

Reply:
(assuming https://github.com/teal-language/tl/pull/1007/files#r2160171826, this could then become

```lua
if toplevel and token.comments then
   store_unattached_comments(node, token, item)
end
```
)